spec/02-integration/11-dbless/03-config_persistence_spec.lua

```diff
@@ -124,31 +124,38 @@ describe("dbless persistence with a declarative config #off", function()
         database = "off",
         declarative_config = yaml_file,
     }))
-    local proxy_client = assert(helpers.proxy_client())
```

Comment:
I remember that the `restart_kong` will wait until all old workers exit and new workers spawn.
Is that not enough to prevent flakiness? Could you explain why this help?

Reply:
`restart_kong` just waits until new workers are spawned, but it doesn't wait until the config is ready.